plugin/pkg/bastion/validator/admission.go

```diff
@@ -150,7 +150,7 @@ func (v *Bastion) Admit(ctx context.Context, a admission.Attributes, _ admission
 	}
 
 	// ensure shoot SSH access is not disabled
-	if shoot.Spec.Provider.WorkersSettings != nil && shoot.Spec.Provider.WorkersSettings.SSHAccess != nil && !shoot.Spec.Provider.WorkersSettings.SSHAccess.Enabled {
+	if shoot.Spec.Provider.WorkersSettings != nil && shoot.Spec.Provider.WorkersSettings.SSHAccess != nil && !shoot.Spec.Provider.WorkersSettings.SSHAccess.Enabled && bastion.DeletionTimestamp == nil {
```

Comment:
Not strictly related to this PR but there is already a helper func that can be reused to simplify the check:
```suggestion
	if bastion.DeletionTimestamp == nil && !helper.ShootEnablesSSHAccess(shoot) {
```